getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,166 @@
+
+    for (likelihood in emotionsLikelihood) { // In this order: VERY_LIKELY, LIKELY, POSSIBLE
+        for (emotion in emotions) { // In this order: JOY, ANGER, SURPRISE, SORROW
+            if (emotion.value == likelihood) return emotion.key // Returns emotion corresponding to likelihood
+        }
+    }
+    return Emoji.NONE
+}
+
+// Downloads source image to a temp file
+fun downloadObject(objectName: String) {
+    val blob = bucket.get(objectName)
+    val localFilePath = Paths.get("/tmp/$objectName")
+    val writeTo = PrintStream(FileOutputStream(localFilePath.toFile()))
+    writeTo.write(blob.getContent())
+    writeTo.close()
+}
+
+class GcsObject(val bucketName: String, val blobName: String)
+class EmojifyResponse(val original: GcsObject, val emojified: GcsObject, val emojifiedUrl: String)
+
+@RestController
+class EmojifyController {
+    @GetMapping("/emojify")
+    fun emojify(@RequestParam(value = "objectName") objectName: String): EmojifyResponse? {
+
+        var publicUrl: String =
+            "https://storage.googleapis.com/${bucket.name}/emojified/emojified-$objectName" // api response
+        val log = Logger.getLogger(Application::class.java.name)
+
+        // Setting up image annotation request
+        val requests = ArrayList<AnnotateImageRequest>()
+        val vision = ImageAnnotatorClient.create()
+        val source = ImageSource.newBuilder().setGcsImageUri("gs://${bucket.name}/$objectName").build() // Fetching our source image
+        val img = Image.newBuilder().setSource(source).build()
+        val feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build()
+        val request = AnnotateImageRequest.newBuilder()
+                .addFeatures(feat)
+                .setImage(img)
+                .build()
+
+        requests.add(request)
+
+        // Calls vision api on above image annotation requests
+        val response = vision.batchAnnotateImages(requests)
+        vision.close()
+
+        for (resp in response.responsesList) {
+            if (resp.hasError()) {
+                log.severe(resp.error.message)
+                return null
+            }
+
+            downloadObject(objectName) // image is now downloaded to /tmp/<objectName>
+            val imgBuff = ImageIO.read(Paths.get("/tmp/$objectName").toFile())
+            val gfx = imgBuff.createGraphics()
+            val extension = (Paths.get("/tmp/$objectName").toFile().extension)
+
+            for (annotation in resp.faceAnnotationsList) {
+                val emoji = bestEmoji(annotation)
+                val imgEmoji = ImageIO.read(Paths.get(EMOJIS_PATH + emojiPic[emoji]).toFile())
+
+                log.info("""
+                    joy: ${annotation.joyLikelihood}
+                    anger: ${annotation.angerLikelihood}
+                    surprise: ${annotation.surpriseLikelihood}
+                    sorrow: ${annotation.sorrowLikelihood}
+                    position: ${annotation.boundingPoly}
+                """)
```

Comment:
It's a nit, but Im bothered that we're logging a giant block like this. It's the first thing my eyes are drawn to when reading through the sample, and it isn't that important. Also, there will be a giant indent (~20 spaces?) in the logs.